src/EditorFeatures/Core/Implementation/IntelliSense/Completion2/Provider.cs

```diff
@@ -0,0 +1,184 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Text;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Language.Intellisense.Prototype.Definition;
+using Prototype = Microsoft.VisualStudio.Language.Intellisense.Prototype.Definition;
+using Microsoft.VisualStudio.Utilities;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Imaging.Interop;
+using Microsoft.VisualStudio.Core.Imaging;
+using Microsoft.CodeAnalysis.Text.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using RoslynCompletionItem = Microsoft.CodeAnalysis.Completion.CompletionItem;
+using RoslynTrigger = Microsoft.CodeAnalysis.Completion.CompletionTrigger;
+using System.Threading;
+using Microsoft.VisualStudio.Language.Intellisense;
+
+namespace RoslynCompletionPrototype
+{
+    [Export(typeof(IAsyncCompletionItemSource))]
+    [Name("C# completion item source")]
+    [ContentType("CSharp")]
+    [ContentType("Basic")]
+    class RoslynCompletionItemSource : IAsyncCompletionItemSource
+    {
+        static readonly ImmutableArray<string> CommitChars = ImmutableArray.Create<string>(".", ",", "(", ")", "[", "]", " ", "\t");
+        const string RoslynItem = nameof(RoslynItem);
```

Comment:
private? Also using typeof(RoslynItem) as a property key is more conventional unless Roslyn has a different convention.